src/pythonfinder/models/mixins.py

```diff
@@ -251,14 +230,14 @@ def next(self) -> Generator:
 
     def find_all_python_versions(
         self,
-        major: str | int | None = None,
-        minor: int | None = None,
-        patch: int | None = None,
-        pre: bool | None = None,
-        dev: bool | None = None,
-        arch: str | None = None,
-        name: str | None = None,
-    ) -> list[PathEntry]:
+        major=None,  # type: Optional[Union[str, int]]
```

Comment:
Is there a reason you used type comments and not type hints directly?

Reply:
This PR came from I made and verified first in pipenv and then differed in from `pipenv` -- it seems master pythonfinder was ahead of what was in pipenv, but only in terms of type annotations.   I corrected many of them, but I couldn't get to all of them.

Reply:
The other thing to note about some of the annotations is they weren't validating under all code paths because sometimes the types used in either a test or one of the paths is slightly different, so I had to loosen their types using Any for the time being.  I figure there will be more iterating on this in the future, especially once pydantic 2 hits a stable release.